src/core/ext/transport/binder/wire_format/binder_android.h

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GRPC_CORE_EXT_TRANSPORT_BINDER_WIRE_FORMAT_BINDER_ANDROID_H_
+#define GRPC_CORE_EXT_TRANSPORT_BINDER_WIRE_FORMAT_BINDER_ANDROID_H_
+
+#if defined(ANDROID) || defined(__ANDROID__)
+
+#include <grpc/impl/codegen/port_platform.h>
+
+#include <android/binder_auto_utils.h>
+#include <android/binder_ibinder.h>
+#include <android/binder_ibinder_jni.h>
+#include <android/binder_interface_utils.h>
+#include <jni.h>
+
+#include <memory>
+
+#include "absl/memory/memory.h"
+#include "src/core/ext/transport/binder/wire_format/binder.h"
+
+// TODO(b/192208764): move this check to somewhere else
+#if __ANDROID_API__ < 29
+#error "We only support Android API level >= 29."
+#endif
+
+namespace binder_transport {
+
+ndk::SpAIBinder FromJavaBinder(JNIEnv* jni_env, jobject binder);
+
+// TODO(mingcl): Find a better name for these classes
```

Comment:
This is an old comment and can be removed I think.

Reply:
Fixed in our internal repo